### ISSUE TITLE
[FEAT] Always indicate spectral subregions for the spectrum-at-apaxel tool

### ISSUE THREAD

**Issue by james-trayford** [spacetelescope/jdaviz#3294]
### Jdaviz component

Cubeviz

### What is the problem this feature will solve?

When selecting a spectral region it can be useful to keep track of how the spectra may change spaxel-to-spaxel in that region. From what I understand these are indicated on the integrated spectrum and also at the `y=0` line in the spectrum viewer, but where the `y=0` line and the integrated spectrum fluxes lie outside the spaxel spectrum's y-axis range (usually below and above the range, respectively), this indication is lost. 

### Describe the desired outcome

I think either having:
- the region indicated as a coloured segment on the spectrum viewer x ***axis*** rather than the `y=0` line
- some low alpha version of the spectral regions designated colour that always shades the plot region between the subselection's wavelength bounds
- Add spectral selection colouring to the spectrum-at-spaxel line - though this is currently red so may clash. It does have a different thickness than the integrated spectrum though, so if changed to black, the selected segments could be coloured in the same way as the integrated spectrum

**Comment by camipacifici** [comment]
Good point! Thank you.
Adding it to the list.

Thinking ahead, if we treat the sonified cubes as data items, we will not need to use the spectrum at spaxel tool anymore. But we might want to still visualize the individual spectra. What do you think?

**Comment by james-trayford** [comment]
Yes, true. I think having these as independent data items would be great - I suppose if you want to see the spectrum change with the sound you ***can*** turn on the SaS tool. As this is working with spectral regions now it should be easier to make it totally independent.
